### ISSUE TITLE
[build_runner 1.11.1] Bad state: Unable to generate package graph

### ISSUE THREAD

**Issue by danallen88** [flutter/flutter#75333]
I am experiencing the same issue as outlined in https://github.com/dart-lang/build/issues/2835 and am tagging @shihaohong per their request.

I followed the instructions to get basic localization working for my project from the [Flutter documentation](https://flutter.dev/docs/development/accessibility-and-localization/internationalization).  I need to use the build_runner for other libraries that require code generation.

## Steps to Reproduce

`build_runner: ^1.11.1`
Running `flutter packages pub run build_runner build -delete-conflicting-outputs` produces the following output:

> $ flutter pub run build_runner build --delete-conflicting-outputs
> Unhandled exception:
> Bad state: Unable to generate package graph, no `/Users/dallen/dev/vcp_diagnostics_mobile/.dart_tool/flutter_gen/pubspec.yaml` found.
> #0      _pubspecForPath (package:build_runner_core/src/package_graph/package_graph.dart:235:5)
> #1      _parsePackageDependencies (package:build_runner_core/src/package_graph/package_graph.dart:208:21)
> #2      PackageGraph.forPath (package:build_runner_core/src/package_graph/package_graph.dart:97:33)
> <asynchronous suspension>
> #3      PackageGraph.forThisPackage (package:build_runner_core/src/package_graph/package_graph.dart:110:20)
> #4      main (file:///Users/dallen/dev/flutter/.pub-cache/hosted/pub.dartlang.org/build_runner-1.11.1/bin/build_runner.dart:27:49)
> #5      _startIsolate.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:299:32)
> #6      _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:168:12)
> pub finished with exit code 255

`flutter doctor`:

> $ flutter doctor
> Doctor summary (to see all details, run flutter doctor -v):
> [✓] Flutter (Channel stable, 1.22.5, on Mac OS X 10.15.7 19H2 darwin-x64, locale en-US)
>  
> [✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
> [✓] Xcode - develop for iOS and macOS (Xcode 11.7)
> [!] Android Studio (version 4.1)
>     ✗ Flutter plugin not installed; this adds Flutter specific functionality.
>     ✗ Dart plugin not installed; this adds Dart specific functionality.
> [✓] IntelliJ IDEA Ultimate Edition (version 2020.2.3)
> [✓] Connected device (1 available)
> 
> ! Doctor found issues in 1 category.

**Comment by darshankawar** [comment]
Added the setup as mentioned in https://flutter.dev/docs/development/accessibility-and-localization/internationalization and was able to replicate the issue on latest stable (1.22.6).

```
dhs@Dhss-MacBook-Pro webtest % flutter packages pub run build_runner build --delete-conflicting-outputs
Unhandled exception:
Bad state: Unable to generate package graph, no `/Users/dhs/Documents/NCFlutter/webtest/.dart_tool/flutter_gen/pubspec.yaml` found.
#0      _pubspecForPath (package:build_runner_core/src/package_graph/package_graph.dart:235:5)
#1      _parsePackageDependencies (package:build_runner_core/src/package_graph/package_graph.dart:208:21)
#2      PackageGraph.forPath (package:build_runner_core/src/package_graph/package_graph.dart:97:33)
<asynchronous suspension>
#3      PackageGraph.forThisPackage (package:build_runner_core/src/package_graph/package_graph.dart:110:20)
#4      main (file:///Users/dhs/documents/Fluttersdk/flutter/.pub-cache/hosted/pub.dartlang.org/build_runner-1.11.1/bin/build_runner.dart:27:49)
#5      _startIsolate.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:299:32)
#6      _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:168:12)
pub finished with exit code 255
dhs@Dhss-MacBook-Pro webtest % 

```

<details>
<summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel stable, 1.22.6, on Mac OS X 10.15.4 19E2269 darwin-x64,
    locale en-IN)
    • Flutter version 1.22.6 at /Users/dhs/documents/Fluttersdk/flutter
    • Framework revision 9b2d32b605 (12 days ago), 2021-01-22 14:36:39 -0800
    • Engine revision 2f0af37152
    • Dart version 2.10.5

 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.0)
    • Android SDK at /Users/dhs/Library/Android/sdk
    • Platform android-30, build-tools 30.0.0
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.3, Build version 12C33
    • CocoaPods version 1.9.3

[!] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.52.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.18.1

[✓] Connected device (3 available)
    • SM A260G (mobile)           • 5200763ebcfa861f                     •
      android-arm • Android 8.1.0 (API 27)
    • sdk gphone x86 arm (mobile) • emulator-5554                        •
      android-x86 • Android 11 (API 30) (emulator)
    • iPhone 12 Pro Max (mobile)  • A5473606-0213-4FD8-BA16-553433949729 • ios
      • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)

! Doctor found issues in 1 category.

```

</details>

**Comment by bitsydarel** [comment]
Same:

`MacBook-Pro-de-darel:flutter_demo darelbitsy$ flutter pub run build_runner build --delete-conflicting-outputs
Unhandled exception:
Bad state: Unable to generate package graph, no `/Users/darelbitsy/IdeaProjects/bd_l10n/example/flutter_demo/.dart_tool/flutter_gen/pubspec.yaml` found.
#0      _pubspecForPath (package:build_runner_core/src/package_graph/package_graph.dart:235:5)
#1      _parsePackageDependencies (package:build_runner_core/src/package_graph/package_graph.dart:208:21)
#2      PackageGraph.forPath (package:build_runner_core/src/package_graph/package_graph.dart:97:33)
<asynchronous suspension>
#3      PackageGraph.forThisPackage (package:build_runner_core/src/package_graph/package_graph.dart:110:20)
#4      main (file:///Users/darelbitsy/tools/flutter/.pub-cache/hosted/pub.dartlang.org/build_runner-1.11.1/bin/build_runner.dart:27:49)
#5      _startIsolate.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:299:32)
#6      _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:168:12)
pub finished with exit code 255
`

flutter doctor:

```
[✓] Flutter (Channel stable, 1.22.6, on macOS 11.2 20D64 darwin-x64, locale
    fr-UA)
 
[!] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    ! Some Android licenses not accepted.  To resolve this, run: flutter doctor
      --android-licenses
[!] Xcode - develop for iOS and macOS
    ✗ Xcode installation is incomplete; a full installation is necessary for iOS
      development.
      Download at: https://developer.apple.com/xcode/download/
      Or install Xcode via the App Store.
      Once installed, run:
        sudo xcode-select --switch /Applications/Xcode.app/Contents/Developer
        sudo xcodebuild -runFirstLaunch
[!] Android Studio
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    ✗ Android Studio not found at /Users/darelbitsy/Library/Application
      Support/JetBrains/Toolbox/apps/AndroidStudio/ch-1/201.6823847/Android
      Studio.app/Contents
[!] Android Studio (version 4.1)
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
[✓] IntelliJ IDEA Ultimate Edition (version 2020.3.2)
[✓] VS Code (version 1.47.3)
[✓] Connected device (1 available)

! Doctor found issues in 4 categories.
```

**Comment by perqin** [comment]
After digging into the source, I found that this issue is caused by `pubspec.yaml` not generated inside the `flutter_gen` package, and has been fixed by this PR #68206. The PR was accepted in v1.24.x, which is still in beta. So you could:
- Switch to beta channel, or
- As a workaround, manually create `.dart_tool/flutter_gen/pubspec.yaml` containing only `name` key

**Comment by shihaohong** [comment]
I'll close this issue for now, since it sounds like @perqin's workaround suffices, and that the `beta` and `master` channels have this problem resolved!
